.github/workflows/nomad-requirements.yml

```diff
@@ -0,0 +1,50 @@
+name: validate NOMAD dependencies compatibility
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  validate_nomad_dependencies:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout pynxtools
+      uses: actions/checkout@v2
+
+    - name: Checkout NOMAD from GitLab
+      run: |
+        git clone https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR.git nomad
+        cd nomad
+        git checkout develop 
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'  # or any other version you need
+
+    - name: Install pip-tools
+      run: |
+        python -m pip install --upgrade pip
+        pip install pip-tools
+
+    - name: Compile pyproject.toml to requirements.txt
+      run: |
+        # Use pip-compile to generate requirements.txt from pyproject.toml
+        pip-compile pyproject.toml --output-file=pyproject-requirements.txt
+        
+    - name: Combine and Check Dependencies
+      run: |
+        # Combine NOMADS requirements with pynxtools dependencies from pyproject.toml
+        cat nomad/requirements-dev.txt > combined-requirements.txt
+        echo "" >> combined-requirements.txt
+        cat pyproject-requirements.txt >> combined-requirements.txt
+
+        # Install the combined dependencies
+        pip install -r combined-requirements.txt
```

Comment:
I think it would make more sense to run nomads pip-compile (or uv) here. Because this is the part which might be failing. Just installing the combined requirements can install other dependencies by packages which are installed later

Reply:
Please have a look at the updated workflow. I switched it such that
- it checks out NOMAD@develop and inits all its submodules
- in nomad/pyproject.toml, it replaces the `pynxtools[convert]==<version>` by `pynxtools[convert]@git+https://github.com/FAIRmat-NFDI/pynxtools.git@<this-current-branch>` to check if the current branch makes any problematic changes
- it then uses NOMADS `uv pip compile` to generate the requirements-(dev).txt
- and then it tries to install them

I checked and the pynxtools version actually does get replaced properly.